sjsonnet/src-jvm-native/sjsonnet/SjsonnetMain.scala

```diff
@@ -7,8 +7,24 @@ import java.nio.file.NoSuchFileException
 import scala.util.Try
 import scala.util.control.NonFatal
 
+// JsonnetWorker (in universe) extends this trait in the universe so that it can pass the cache based on Caffeine to main0 here
+trait ParseCacheInterface {
+  def getOrElseUpdate(key: (Path, String), defaultValue:  => Either[Error, (Expr, FileScope)]): Either[Error, (Expr, FileScope)]
+}
+
+// As JsonnetWorker (in universe) passes an object to main0 which extends the trait: ParseCacheInterface, class based on HashMap
+// which extends this trait is needed to use the default HashMap
+case class HashMapDefault () extends ParseCacheInterface {
+  val cache = new collection.mutable.HashMap[(Path, String), Either[Error, (Expr, FileScope)]]()
+
+  override def getOrElseUpdate(key: (Path, String), defaultValue:  => Either[Error, (Expr, FileScope)]): Either[Error, (Expr, FileScope)] = {
+    cache.getOrElseUpdate((key._1, key._2), defaultValue)
+  }
+}
+
 object SjsonnetMain {
-  def createParseCache() = collection.mutable.HashMap[(Path, String), Either[Error, (Expr, FileScope)]]()
+//  def createParseCache() = collection.mutable.HashMap[(Path, String), Either[Error, (Expr, FileScope)]]()
+  def createParseCache() = new HashMapDefault
```

Comment:
The presence or absence of an argument list at the call site should generally match the definition (and this is often required in Scala). In this case it's better to change the definition (`class DefaultParseCache` instead of `class DefaultParseCache()`). Since it's a constructor, you already know that it always allocates a new object, and there are no further side-effects.